client/src/stores/IpfsDaemonStore.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
```

Comment:
We should refactor the IpfsDaemonStore a bit. The idea is to keep the daemon instance in this store, not the settings (although settings can be part of it too).

So at the basic level, we should have `IpfsDaemonStore.ipfs` which would give us the IPFS instance (daemon).

To do this, we initialize the variable in `init()` with `this.ipfs = null`. When the daemon has been started, we set it to `this.ipfs = ipfsInstance`. To do this, we need to refactor the actions and callbacks a little.

I would move the `ipcRenderer.once('ipfs-daemon-instance', ...` line to init() first. Then, let's create appropriate actions to handle the state of the ipfs instance: `IpfsDaemonActions.start` (which is called when the UI wants to start the daemon) and `IpfsDaemonActions.daemonStarted` which is called when this store has received and IPFS instance:
```javascript
     ipcRenderer.once('ipfs-daemon-instance', () => {
        logger.info('daemon callback')
        this.ipfs = remote.getGlobal('ipfsInstance')
        IpfsDaemonActions.daemonStarted(this.ipfs)
      })
```

That we we can decouple the different states and stores from each other. This allows us to handle the "IPFS daemon was started" event in other place. I would put this in the "App", ie. in App you listen for IpfsDaemonActions `onDaemonStarted` and can call `NetworkActions.connect(null, username);` there (instead of in LoginView).